TruePath.Tests/PathStringsTests.cs

```diff
@@ -62,17 +62,55 @@ public void DotFoldersAreTraversed(string input, string expected)
         Assert.Equal(NormalizeSeparators(expected), PathStrings.Normalize(input));
     }
 
-    [Theory(Skip = "TODO[#16]: Make it work properly")]
-    [InlineData("C:.", "C:")]
-    [InlineData("C:./foo", "C:foo")]
-    [InlineData("C:..", "C:..")]
-    [InlineData("C:/..", "C:/..")]
+    [Theory]
+    [InlineData(".", "")]
+    [InlineData("./foo", "foo")]
+    [InlineData("..", "..")]
+    [InlineData("./..", "..")]
+    [InlineData("a/..", "")]
+    [InlineData("a/../..", "..")]
+    [InlineData("a/../../.", "..")]
+    [InlineData("a/../../..", "../..")]
+    [InlineData("foo/./bar/../var/./dar/..", "foo/var")]
+    [InlineData("foo/.bar", "foo/.bar")]
+    [InlineData("/.", "/")]
+    [InlineData("/..", "/..")]
+    [InlineData("/../..", "/../..")]
+    [InlineData("/../../foo/..", "/../..")]
+    [InlineData("x/foo/bar/../..", "x")]
+    [InlineData("x/foo/bar/.../.", "x/foo/bar/...")]
+    [InlineData("x/foo/..bar/", "x/foo/..bar")]
+    [InlineData("../../foo", "../../foo")]
+    [InlineData("../../../foo", "../../../foo")]
+    [InlineData("../foo/..", "..")]
     public void WindowsSpecificDotFoldersAreTraversed(string input, string expected)
     {
         if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) return;
-        Assert.Equal(NormalizeSeparators(expected), PathStrings.Normalize(input));
+
+        // Arrange
+        var driveLetter = RandomDriveLetter();
+        input = driveLetter + input;
+        expected = driveLetter + expected;
+
+        //Act
+        var actual = PathStrings.Normalize(input);
+
+        // Assert
+        Assert.Equal(NormalizeSeparators(expected), actual);
     }
 
     private static string NormalizeSeparators(string path) =>
         path.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
+
+    private static string RandomDriveLetter()
+    {
+        var rnd = new Random(Guid.NewGuid().GetHashCode());
```

Comment:
I'd prefer us to use a static `Random` instance with hard-coded seed, for tests to be reproducible.

Or even get away from `Random` and just append `C:` everywhere.

Reply:
The main purpose of this test is to make sure that any drive letter is ignored. To do this, we have created a test in the style of Monkey Testing, where we don't care about the drive letter and take any of them and expect the same result. Why is there a C or other drive letter?

Reply:
Say we have somewhow missed the letter `X`. The CI detects it anmd reports a failure, and to fix it, the developer needs to verify that `X` is now supported. How do they do that, if the tests are random and generate other input on the developer's computer?

Reply:
ITestOutputHelper. You can always detect a random seed or drive letter.

Reply:
I still don't like it, but whatever.